gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MFAEnrollStep.java

```diff
@@ -52,8 +51,9 @@ public MFAEnrollStep(Handler<RoutingContext> wrapper, RuleEngine ruleEngine, Fac
     @Override
     public void execute(RoutingContext routingContext, AuthenticationFlowChain flow) {
         final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
-        final MfaFilterContext context = new MfaFilterContext(routingContext, client, factorManager);
+        final MfaFilterContext context = new MfaFilterContext(routingContext, client, factorManager, ruleEngine);
         if (hasFactors(client, factorManager)) {
+            context.setDefaultFactorWhenApplied();
```

Comment:
issue: I still thinking this "setDefault" is more something to manage in the EnrollmenetEndpoint on the GET method to provide the default factor if rules don't provide at least one factor, another check in the EnrollmentEndpoint on the POST method can be done in order to confirm the user is enrolling a factor restricted by the SelectionRules/or the default.

Reply:
we have a story for this. Let's discuss and continue within [AM-1138](https://gravitee.atlassian.net/browse/AM-1138)